### PR TITLE
Project Restructuring: Introduction of 'backend', 'frontend', 'common' Directories

### PR DESCRIPTION
**Pull request by FlynnOConnell**
We have restructured our project to improve modularity and clarity by introducing 'backend', 'frontend', and 'common' directories.

1. Renamed 'rocket-api' to 'backend' directory.
2. Moved all content from 'yew-ui' to 'frontend/yew-ui' directory.
3. Updated the workspace member paths in the root 'Cargo.toml' to reflect the new structure.
4. Adjusted import paths in Rust files for compatibility with the new structure.
5. Added boilerplate in 'common' directory for potential data type implementations.

No code functionality has been altered, it solely focuses on improving our codebase's structure to enhance clarity, future development, and ease of navigation.

### CODE REVIEW

File: backend/rocket-api/Cargo.toml [hunk 1]
```diff
@@ -6,5 +6,5 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-rocket = "=0.5.0-rc.3"
-common = { path = "../common" }
\ No newline at end of file
+rocket = "0.4.11""
```
Comment (FlynnOConnell) [inline backend/rocket-api/Cargo.toml]: oops, fixed [a59c30f](https://github.com/LukeBrandon/flyuke/pull/9/commits/a59c30f4d5cdd7ac1d630e32beb8cdbff3da3bda)

File: backend/rocket-api/Cargo.toml [hunk 2]
```diff
@@ -6,4 +6,5 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-rocket = "=0.5.0-rc.3"
```
Comment (FlynnOConnell) [inline backend/rocket-api/Cargo.toml:9]: ew my brother was on my computer and did this 

fixed: [6150ebe](https://github.com/LukeBrandon/flyuke/pull/9/commits/6150ebec35aacac424358373bb703c416982240f)

File: frontend/.github/workflows/pr_check.yml
Comment (LukeBrandon) [inline frontend/.github/workflows/pr_check.yml:1]: The `yew-ui` will be all there is to the front end, so adding another `frontend` directory is redundant.  The `yew-ui` structure was the boilerplate app provided by yew, so I wouldn't change this at all.

File: backend/rocket-api/src/main.rs [hunk 3]
```diff
@@ -1,5 +1,7 @@
 #[macro_use]
 extern crate rocket;
+use common::User;
+use common::Pet;
```
Comment (LukeBrandon) [inline backend/rocket-api/src/main.rs]:
```suggestion
```
Remove unused `use` statements

File: backend/rocket-api/Cargo.toml (hunk 2 above)
Comment (LukeBrandon) [reply backend/rocket-api/Cargo.toml:9]: :)

File: backend/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+tokio = { version = "1.16.1", features = ["full"] }
+serde = { version = "1.0.130", features = ["derive"] }
+serde_json = "1.0.68"
+# warp is optimized for tokio
+warp = "0.3.1"
```
Comment (LukeBrandon) [inline backend/Cargo.toml]: why is `warp` being added here?

File: common/src/lib.rs
Comment (LukeBrandon) [inline common/src/lib.rs:1]: Where are these structs from? Just random stuff?

File: common/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "common"
```
Comment (LukeBrandon) [inline common/Cargo.toml:2]: what is the purpose of this going to be? To share types?

**Review by LukeBrandon — CHANGES_REQUESTED** [review]
Lets not over complicate things for now, just get things moving and then we can figure out how to better structure things once we know more about rust and rust structure that makes sense.

File: yew-ui/index.html
Comment (LukeBrandon) [inline yew-ui/index.html:1]: There is also another `yew-ui` just sitting here

File: common/Cargo.toml (hunk 5 above)
Comment (FlynnOConnell) [reply common/Cargo.toml:2]: Common types, shared functions, error handling, database connection strings, secrets, database schema

File: common/src/lib.rs
Comment (FlynnOConnell) [reply common/src/lib.rs:1]: Completely random stuff just to have something to initialize and test

File: frontend/.github/workflows/pr_check.yml
Comment (FlynnOConnell) [reply frontend/.github/workflows/pr_check.yml:1]: I think removing the yew-ui directory and just having src makes sense, but I think keeping the general frontend directory makes a lot of sense and isn't over complicating anything

File: backend/Cargo.toml (hunk 4 above)
Comment (FlynnOConnell) [reply backend/Cargo.toml]: Define routes, http requests, process requests and interact with the db

Comment (LukeBrandon) [reply backend/Cargo.toml]: that's what `rocket` is for, `warp` is a competitor to rocket lol

File: common/Cargo.toml (hunk 5 above)
Comment (LukeBrandon) [reply common/Cargo.toml:2]: Common types make sense, but anything DB related will be a BE thing

File: Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+[workspace]
+
+members = [
+    "backend/rocket-api",
+    "frontend/yew-ui",
```
Comment (LukeBrandon) [inline Cargo.toml]: need to update this after moving the crates around

**Review by LukeBrandon — CHANGES_REQUESTED** [review]
rename `rocket-api` to backend and delete the outer `backend` crate, redunant.

File: backend/rocket-api/Cargo.toml [hunk 7]
```diff
@@ -6,4 +6,5 @@ edition = "2021"
...
\ No newline at end of file
+rocket = "0.4.11"
+common = { path = "../../common" }
```
Comment (LukeBrandon) [inline backend/rocket-api/Cargo.toml]:
```suggestion
common = { path = "../common" }
```

File: backend/Cargo.toml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+tokio = { version = "1.16.1", features = ["full"] }
+serde = { version = "1.0.130", features = ["derive"] }
+serde_json = "1.0.68"
+sqlx = { version = "0.6.0", features = ["rt-tokio"] }
```
Comment (LukeBrandon) [inline backend/Cargo.toml]:
```suggestion
```

**Review by LukeBrandon — APPROVED** [review]

**Pull request merged**